TeaDiary/Scripts/App/TeaCommands.js

```diff
@@ -0,0 +1,46 @@
+﻿function TeaCommands() {
+    var self = this;
+    self.delete = (id, successCallback, errorCallback) => {
+        popupConfirmation(
+            {
+                 body: "Are you sure you want to delete this tea? You cannot undo this action"
+            },
+            () => apiCommands.deleteTea(id, successCallback, errorCallback)
+        );
+    }
+
+    function popupConfirmation(content, confirmCallback, rejectCallback) {
+        content = popupConfirmationContentValidation(content);
+        if (confirm(content.body)) {
+            confirmCallback();
+        } else {
+            rejectCallback();
+        }
+    }
+
+    function popupConfirmationContentValidation(content) {
```

Comment:
It's not a validation. It's providing the default values for your configuration.
Do it like this:
```
myMethod(params) {
const default = { ... }; // your Defaults
var config = Object.assign({}, default, params);
...
// use config in your code
}
```